README.md

```diff
@@ -386,6 +384,82 @@ the ownership voucher and ownership certificate.
         we have validated the identity and integrity of the device and its
         software components. It is ready to serve traffic.
 
+### Bootz Procedure (BootstrapStream)
+
+1. DHCP Discovery of Bootstrap Server
+    1. Device sends DHCP messages, containing the mac-address of the active
+        control card. The DHCP server has been configured with all possible
+        mac-addresses of the device, and responds with the static IP address of
+        the bootstrap server.
+    2. DHCP server also assigns an IP address and a gateway to the device.
+    3. The DHCP response option code should be OPTION_V4_SZTP_REDIRECT(143) or
+        OPTION_V6_SZTP_REDIRECT(136).
+    4. The format of the DHCP message (other than response option code) follows
+        [RFC](https://www.rfc-editor.org/rfc/rfc8572#page-56).
+        1. The URI will be in the format of bootz://&lt;host or ip>:&lt;port>
+2. Bootstrapping Service
+    1. Device initiates a gRPC connection to the bootz-server whose address was
+        obtained from the DHCP server.
+    2. The TLS connection may use any certificate it has available. This can
+        be a self-signed certificate as the cert is only used to encrypt the
+        connection.
+    4. The responses from the bootz-server are signed by ownership-certificate.
+        The device validates the ownership-voucher, which authenticates the
+        ownership-certificate. The device verifies the signature of the message
+        body before accepting the message.
+    5. If the signature could not be verified, the bootstrap process starts
+        from Step 1.
+2. BootstrapStreamRequest.bootstrap_request
+    1. The bootz BootstrapStream is initiated by the device sending a
+        GetBootstrappingData message to the bootz-server in the first message
+        of the stream.
+    3. The device describes itself, by listing out all available control cards,
+        and their states.
+    4. For a full device install, the state of all control cards is
+        NOT\_INITIALIZED.
+    5. For installing hot-swapped modules (RMA), the primary control card sets
+        its state to INITIALIZED, and that of the swapped module to
+        NOT\_INITIALIZED.
+3. BootstrapStreamResponse.challenge
+    1. The server will provide a challenge depending on the data fetched from
+        OVGS server based on the GetBootstrapDataRequest.identity.
+    2. For TPM2.0 with iDevID
+       1. The iDevID cert will be validated and the S/N will
```

Comment:
Maybe a gap in my knowledge: what does the iDevID cert being "validated" mean? Is this checking that the `idevid_cert` that is supplied in the `GetBootstrapDataRequest` is the same certificate as the device thinks that it has?